### PR TITLE
Refactor Game construction and move some tests

### PR DESCRIPTION
**Pull request by kiliangrashoff**
Game is now instantiated using a GameFactory, which right now is
trivial, but when our object creation graph improves, it can
create all the objects that are used in a game instead of them
instantiating their dependencies themselves. This will improve
our ability to test classes independently.

This commit also moves Piece and Gem tests to the correct package.

< organisation, interface

### CODE REVIEW

**Comment by ghyl** [comment]
lgtm

File: src/main/java/nl/tudelft/ti2206/semgr8/model/Game.java [hunk 1]
```diff
@@ -19,10 +17,8 @@
     /**
      * Constructor that generates a new board.
      */
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/Game.java]: java doc not yet updated

File: src/main/java/nl/tudelft/ti2206/semgr8/model/GameFactory.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package nl.tudelft.ti2206.semgr8.model;
+
+/**
+ * Constructs a game with the objects it needs.
+ */
+public class GameFactory {
+    public Game getGame() {
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/GameFactory.java]: missing javadoc

**Comment by albert-smit** [comment]
looks good besides missing javadoc

**Comment by wroel** [comment]
this pull request is also in pr #54

**Pull request closed without merging**